Investigate test_indexer test failure in Travis build #6228

test_indexer failed in Travis build #6228:
https://travis-ci.org/HumanCellAtlas/data-store/builds/290603179

The failed test output is as follows:
```
======================================================================
FAIL: test_subscription_notification_successful (tests.test_indexer.TestGCPIndexer)
----------------------------------------------------------------------
Traceback (most recent call last):
 File "/home/travis/build/HumanCellAtlas/data-store/tests/test_indexer.py", line 244, in test_subscription_notification_successful
 self.verify_notification(subscription_id, smartseq2_paired_ends_query, bundle_id)
 File "/home/travis/build/HumanCellAtlas/data-store/tests/test_indexer.py", line 276, in verify_notification
 self.assertEqual(subscription_id, posted_json['subscription_id'])
AssertionError: 'c6cfc4e6-88e7-4454-8635-631d808a0eb4' != '72b31d83-661b-44f2-b242-68c14dd345a9'
- c6cfc4e6-88e7-4454-8635-631d808a0eb4
+ 72b31d83-661b-44f2-b242-68c14dd345a9
```

Other Travis builds with the same code have passed, this failure has only occurred once.

This appears to be due to using the notification test handler twice in the same test without resetting it between uses. The notification test handler is reset before each test by setUp calling PostTestHandler.reset(). However, when multiple notifications are performed within a single test, the test notification handler needs to be reset explicitly by calling this same method, PostTestHandler.reset(). Not doing so results in a test race condition, which explains the intermittent failure of this test. This is evidenced by the following test output, indicating that the unexpected subscription id value is actually that of the previous notification performed within this same test:
```
INFO:tests.test_indexer:Received gcp creation event for bundle which will be indexed: bundles/4fffdc18-8f3c-499c-b828-512ffac0e99f.2017-10-20T194654.629046Z
127.0.0.1 - - [20/Oct/2017 19:46:58] "POST / HTTP/1.1" 200 -
INFO:tests.test_indexer:Successfully notified for subscription 72b31d83-661b-44f2-b242-68c14dd345a9 for bundle 4fffdc18-8f3c-499c-b828-512ffac0e99f.2017-10-20T194654.629046Z with transaction id a6f815dc-af5f-4128-98e3-368ae0b9b6bd Code: 200
INFO:connexion.api.security:... Token authenticated.
INFO:connexion.api.security:... Token authenticated.
INFO:tests.test_indexer:Received gcp creation event for bundle which will be indexed: bundles/56f340e1-c8b9-4bef-9416-57afe91df761.2017-10-20T194705.124327Z
127.0.0.1 - - [20/Oct/2017 19:47:08] "POST / HTTP/1.1" 200 -
INFO:tests.test_indexer:Successfully notified for subscription c6cfc4e6-88e7-4454-8635-631d808a0eb4 for bundle 56f340e1-c8b9-4bef-9416-57afe91df761.2017-10-20T194705.124327Z with transaction id da9a1c99-cec3-4f10-a2e6-698af59cc0ad Code: 200
F[2017-10-20T19:47:08,365][INFO ][o.e.c.m.MetaDataDeleteIndexService] [HJgz4tE] [dss-docs-gcp-dev-test_indexer/qwUoTT12QayAO7F5Bm8MoA] deleting index
[2017-10-20T19:47:08,366][INFO ][o.e.c.m.MetaDataDeleteIndexService] [HJgz4tE] [dss-subscriptions-gcp-dev-test_indexer/JyF4BMvTT7aZ_9jo3XVSFQ] deleting index
INFO:tests.test_indexer:Received gcp creation event for bundle which will be indexed: bundles/1d65956e-b2fa-4779-a6a2-dda4caadf5ea.2017-10-20T194715.654512Z
[2017-10-20T19:47:18,667][INFO ][o.e.c.m.MetaDataCreateIndexService] [HJgz4tE] [dss-docs-gcp-dev-test_indexer] creating index, cause [api], templates [], shards [5]/[1], mappings [query]
[2017-10-20T19:47:20,090][INFO ][o.e.c.m.MetaDataMappingService] [HJgz4tE] [dss-docs-gcp-dev-test_indexer/WQamgWSaRAGeHyM8DfNPOA] create_mapping [doc]
[2017-10-20T19:47:20,211][INFO ][o.e.c.m.MetaDataCreateIndexService] [HJgz4tE] [dss-subscriptions-gcp-dev-test_indexer] creating index, cause [api], templates [], shards [5]/[1], mappings []
INFO:connexion.api.security:... Token authenticated.
[2017-10-20T19:47:20,594][INFO ][o.e.c.m.MetaDataMappingService] [HJgz4tE] [dss-subscriptions-gcp-dev-test_indexer/Uw3oYxL0QKuBoBhWHbjjpg] create_mapping [subscription]
127.0.0.1 - - [20/Oct/2017 19:47:32] "POST / HTTP/1.1" 500 -
.
======================================================================
FAIL: test_subscription_notification_successful (tests.test_indexer.TestGCPIndexer)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/HumanCellAtlas/data-store/tests/test_indexer.py", line 244, in test_subscription_notification_successful
    self.verify_notification(subscription_id, smartseq2_paired_ends_query, bundle_id)
  File "/home/travis/build/HumanCellAtlas/data-store/tests/test_indexer.py", line 276, in verify_notification
    self.assertEqual(subscription_id, posted_json['subscription_id'])
AssertionError: 'c6cfc4e6-88e7-4454-8635-631d808a0eb4' != '72b31d83-661b-44f2-b242-68c14dd345a9'
- c6cfc4e6-88e7-4454-8635-631d808a0eb4
+ 72b31d83-661b-44f2-b242-68c14dd345a9
```
This should be resolved simply by adding a call to PostTestHandler.reset() between the notifications performed by this test.

--- a/tests/test_indexer.py
+++ b/tests/test_indexer.py
@@ -243,6 +243,7 @@ class TestIndexerBase(DSSAssertMixin, DSSStorageMixin, DSSUploadMixin):
             prefix, _, bundle_id = bundle_key.partition("/")
             self.verify_notification(subscription_id, smartseq2_paired_ends_query, bundle_id)
             self.delete_subscription(subscription_id)
+            PostTestHandler.reset()
 
     def test_subscription_notification_unsuccessful(self):
         PostTestHandler.verify_payloads = True

